src/ops/inc_multiquery_attention.cc

```diff
@@ -0,0 +1,1430 @@
+/* Copyright 2023 CMU, Facebook, LANL, MIT, NVIDIA, and Stanford (alphabetical)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "flexflow/ops/inc_multiquery_attention.h"
+#include "flexflow/ffconst_utils.h"
+#include "flexflow/model.h"
+#if defined(FF_USE_CUDA) || defined(FF_USE_HIP_CUDA)
+#include "flexflow/utils/cuda_helper.h"
+#else
+#include "flexflow/utils/hip_helper.h"
+#endif
+#include "flexflow/utils/hash_utils.h"
+#include "legion/legion_utilities.h"
+#ifdef INFERENCE_TESTS
+#include <torch/torch.h>
+using namespace at::indexing;
+#endif
+
+namespace FlexFlow {
+
+// declare Legion names
+using Legion::ArgumentMap;
+using Legion::Context;
+using Legion::coord_t;
+using Legion::Domain;
+using Legion::FutureMap;
+using Legion::IndexLauncher;
+using Legion::Machine;
+using Legion::Memory;
+using Legion::PhysicalRegion;
+using Legion::Predicate;
+using Legion::Rect;
+using Legion::RegionRequirement;
+using Legion::Runtime;
+using Legion::Task;
+using Legion::TaskArgument;
+using Legion::TaskLauncher;
+using PCG::Node;
+
+LegionRuntime::Logger::Category log_inc_mqa("IncrementalMQA");
+
+bool IncMultiQueryAttentionParams::is_valid(
+    ParallelTensorShape const &input) const {
+  bool is_valid = input.is_valid();
+  return is_valid;
+}
+
+Tensor FFModel::inc_multi_query_attention(const Tensor input,
```

Comment:
inc_multi_query_attention -> inc_multiquery_self_attention. This is to align with our naming convention for MHA.